### PR TITLE
Letting perfproc errors pass silently

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/19474

Unfortunately a reliable detection if performance process counters are enabled isn't possible. Therefore we need to let them pass silently.

Not a pretty fix but I can't see any other way around that.

### CODE REVIEW

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 1]
```diff
@@ -940,15 +941,30 @@ private static bool ProcessPerformanceCounterEnabled()
...
-            Assert.Equal(currentProcess.EnableRaisingEvents, remoteProcess.EnableRaisingEvents);
-            Assert.Equal("127.0.0.1", remoteProcess.MachineName);
-            // This property throws exception only on remote processes.
-            Assert.Throws<NotSupportedException>(() => remoteProcess.MainModule);
+        public void TestProcessOnRemoteMachineWindows()
+        {
+            Process currentProcess = Process.GetCurrentProcess();
+
+            void TestRemoteProccess(Process remoteProcess)
+            {
```
Comment (Priya91) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: C# supports local functions now! didn't know :)

Comment (ViktorHofer) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: Yes 👍

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 2]
```diff
@@ -878,7 +878,15 @@ public void GetProcessesByName_ProcessNameMachineName_ReturnsExpected(string mac
         [PlatformSpecific(TestPlatforms.Windows)] // Accessing processes on remote machines is only supported on Windows.
         public void GetProcessesByName_RemoteMachineNameWindows_ReturnsExpected(string machineName)
         {
-            GetProcessesByName_ProcessNameMachineName_ReturnsExpected(machineName);
+            try
+            {
+                GetProcessesByName_ProcessNameMachineName_ReturnsExpected(machineName);
+            }
+            catch (InvalidOperationException)
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: We have both this and the ConditionalTheory?

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 3]
```diff
@@ -878,7 +878,15 @@ public void GetProcessesByName_ProcessNameMachineName_ReturnsExpected(string mac
...
+            {
+                // As we can't detect reliably if performance counters are enabled 
+                // we let possible InvalidOperationExceptions pass silently.
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: Why can't we detect it reliably?  Have we followed up with an expert in Windows who could help?

Comment (ViktorHofer) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: @danmosemsft do you know somebody from the Windows team who could help?

Comment (danmosemsft) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: IM"d you that.

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 4]
```diff
@@ -940,15 +941,30 @@ private static bool ProcessPerformanceCounterEnabled()
...
-            Assert.Equal(currentProcess.BasePriority, remoteProcess.BasePriority);
-            Assert.Equal(currentProcess.EnableRaisingEvents, remoteProcess.EnableRaisingEvents);
-            Assert.Equal("127.0.0.1", remoteProcess.MachineName);
-            // This property throws exception only on remote processes.
-            Assert.Throws<NotSupportedException>(() => remoteProcess.MainModule);
+        public void TestProcessOnRemoteMachineWindows()
+        {
+            Process currentProcess = Process.GetCurrentProcess();
+
+            void TestRemoteProccess(Process remoteProcess)
```
Comment (danmosemsft) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: typo 'proccess'

**Comment by ViktorHofer** [comment]
@stephentoub After speaking with @danmosemsft, I removed the registry check for now. To find a reliable detection I follow up with the corresponding team per mail (right now) and CC you.

**Pull request merged**